### PR TITLE
refactor(woker): add function to set fail status for event

### PR DESCRIPTION
**Pull request by supereagle**
## Fixes [ISSUE-XXX](issue-link-address)

**Changes**
  - [x] Refactor: add function to set fail status for event

### CODE REVIEW

**Comment by superxi911** [comment]
Do you have finished this PR？

**Comment by supereagle** [comment]
Yes, just add a function for common code.

File: worker/worker.go [hunk 1]
```diff
@@ -358,3 +337,10 @@ func isChanClosed(ch chan interface{}) bool {
 	}
 	return false
 }
+
+// setEventFailStatus sets the fail status of the event.
+func setEventFailStatus(event *api.Event, ErrorMessage string) {
+	event.Status = api.EventStatusFail
+	event.ErrorMessage = ErrorMessage
+	logdog.Error("Operation failed", logdog.Fields{"event": event})
+}
```
Comment (superxi911) [inline worker/worker.go]: needs a new line

Comment (supereagle) [reply worker/worker.go]: Done

**Pull request merged**